import Joi from "joi";
import generateError from "../../utils/generateError.js";
import {
  getUserByVerificationCode,
  updateVerificationStatus,
} from "../../models/users/index.js";

const verify = async (req, res, next) => {
  try {
    const { verificationCode } = req.body;

    const schema = Joi.object().keys({
      email: Joi.string().email().required(),
      verificationCode: Joi.string().required(),
    });

    const validation = schema.validate(req.body);

    if (validation.error) {
      generateError(validation.error.message, 400);
    }

    const user = await getUserByVerificationCode(verificationCode);

    if (!user) {
      generateError("C贸digo de verificaci贸n incorrecto.", 400);
    }

    const status = await updateVerificationStatus(email);

    res.status(200).send({
      message: "Verificaci贸n exitosa. Ahora puedes iniciar sesi贸n.",
    });
  } catch (error) {
    next(error);
  }
};

export default verify;
